controllers/managedosversionchannel_controller.go

```diff
@@ -357,6 +359,29 @@ func (r *ManagedOSVersionChannelReconciler) createManagedOSVersions(ctx context.
 			vcpy.Spec.UpgradeContainer = ch.Spec.UpgradeContainer
 		}
 
+		if len(ch.Spec.Registry) > 0 {
+			registry := ch.Spec.Registry
+			urlField := ""
+			switch {
+			case vcpy.IsContainerImage():
+				urlField = "upgradeImage"
+			case vcpy.IsISOImage():
+				urlField = "uri"
+			default:
+				err := fmt.Errorf("unexpected ManagedOSVersion type %s", vcpy.Spec.Type)
+				logger.Error(err, "failed to concatenate managedosversion with registry", "registry", ch.Spec.Registry)
+				return err
+			}
+
+			val := strings.Trim(string(vcpy.Spec.Metadata[urlField].Raw), "\"")
+			if !strings.HasSuffix(ch.Spec.Registry, "/") && !strings.HasPrefix(val, "/") {
+				registry += "/"
+			}
+			vcpy.Spec.Metadata[urlField] = runtime.RawExtension{
```

Comment:
runtime.RawExtension direct manipulation of "Raw" Data: not so proud of this, but going through multiple rounds of Marshal/Unmarshal of the data to just change that single value looks overkill 🤷🏼

Reply:
Indeed, but to be fair this is an API problem, same for `upgradeImage` vs `uri` :D
`v1beta2` is something we should start thinking about